src/com/maddyhome/idea/vim/group/ChangeGroup.java

```diff
@@ -1910,6 +1910,56 @@ private void exitAllSingleCommandInsertModes(@NotNull Editor editor) {
     }
   }
 
+  /**
+   * Deletes next search, or current if cursor is on a match.
+   *
+   * @param editor  The editor to insert the text into
+   * @param caret
+   * @param context The data context
+   * @param count   The number times to search before deleting
+   * @param forwards
+   * @return true if able to find and delete the search, false if not
+   */
+  public boolean deleteNextSearch(Editor editor, Caret caret, DataContext context, int count, boolean forwards) {
+    String lastSearch = VimPlugin.getSearch().getLastSearch();
+    int currentPos = editor.getCaretModel().getOffset();
+    TextRange nextRange = new TextRange(currentPos,currentPos);
+    for(int i = 0; i< count; i++) {
+      nextRange = VimPlugin.getSearch().findCurrentOrNext(editor, lastSearch, nextRange.getStartOffset(), false, forwards);
+      if (nextRange == null) {
+        return false;
+      }
+    }
+    MotionGroup.moveCaret(editor,caret, nextRange.getStartOffset());
+    return VimPlugin.getChange().deleteRange(editor, caret, new TextRange(nextRange.getStartOffset(),nextRange.getEndOffset()),SelectionType.CHARACTER_WISE, true);
+  }
+  /**
+   * Change next search, or current if cursor is on a match.
+   *
+   * @param editor  The editor to insert the text into
+   * @param context The data context
+   * @param count   The number times to search before deleting
+   * @param forwards
+   * @return true if able to find and delete the search, false if not
+   */
+  public boolean changeNextSearch(Editor editor, DataContext context, int count, boolean forwards) {
+    String lastSearch = VimPlugin.getSearch().getLastSearch();
+    int currentPos = editor.getCaretModel().getOffset();
+    TextRange nextRange = new TextRange(currentPos,currentPos);
+    for(int i = 0; i< count; i++) {
+      nextRange = VimPlugin.getSearch().findCurrentOrNext(editor, lastSearch, nextRange.getStartOffset(), false, forwards);
+      if (nextRange == null) {
+        return false;
+      }
+    }
+
+    for (Caret caret : editor.getCaretModel().getAllCarets()) {
```

Comment:
IJ reports that "this is not a loop" because it has always only one iteration. Is here everything fine?

Reply:
Yes, I missed that one. Changed.